cache/cache.go

```diff
@@ -27,3 +27,8 @@ type Item struct {
 	Value  interface{}
 	Expiry time.Time
 }
+
+// HasExpired returns true if the item has expired.
+func (i *Item) HasExpired() bool {
```

Comment:
I think this a problem for the item. It needs to be exposed as well so it can be used by the cache. 

I personally think it makes the API simpler for someone implementing another cache. As they will have to use the Item struct to satisfy the Cache interface. So it would mean they don't need to worry about implementing how to check if the item is expired. 

But thinking on it a bit more, it would make future updates easier for us by not having to support that method on the item.